sherpa/astro/tests/test_astro_data_swift_unit.py

```diff
@@ -109,6 +101,137 @@ def test_read_pha(make_data_path):
     assert pha.background_ids == []
     assert pha.get_background() is None
 
+    # Select a few header keywords. Some are OGIP related,
+    # some are "we expect them", and some are there just to
+    # check we encode the type / value correctly.
+    #
+    assert pha.header["HDUCLASS"] == "OGIP"
+    assert pha.header["HDUCLAS1"] == "SPECTRUM"
+    assert pha.header["HDUCLAS2"] == "TOTAL"
+    assert pha.header["HDUCLAS3"] == "COUNT"
+    assert pha.header["DETCHANS"] == 1024
+    assert pha.header["CHANTYPE"] == "PI"
+
+    # This keyword is removed from the header.
+    assert "POISERR" not in pha.header
+
+    assert pha.header["TELESCOP"] == "SWIFT"
+    assert pha.header["INSTRUME"] == "XRT"
+    assert pha.header["FILTER"] == "NONE"
+    assert pha.header["OBJECT"] == "2M02594633+3855363"
+
+    assert pha.header["RA_OBJ"] == pytest.approx(44.9445)
+    assert pha.header["DEC_OBJ"] == pytest.approx(38.9263)
+    assert pha.header["UTCFINIT"] == pytest.approx(-15.64118)
+
+    assert not pha.header["XBADFRAM"]
+
+
+def check_arf(arf):
+    """Does the ARF contain the expected values?"""
+
+    assert isinstance(arf, DataARF)
+
+    nbins = 2400
+
+    # Expect the upper edge of bin j to equal the lower
+    # edge of bin j + 1. Note that we check that the low
+    # bin has been relpaced (i.e. is not 0.0).
```

Comment:
```suggestion
    # bin has been replaced (i.e. is not 0.0).
```

Reply:
I actually just found and fixed that in my version. I am going to be doing significant work on this PR so I don't think it's worth much review just yet, as there's plenty of places for me to add typos!